x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/sagemaker/schema/SageMakerStreamSchemaPayload.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.sagemaker.schema;
+
+import software.amazon.awssdk.core.SdkBytes;
+
+import org.elasticsearch.inference.InferenceServiceResults;
+import org.elasticsearch.inference.TaskType;
+import org.elasticsearch.inference.UnifiedCompletionRequest;
+import org.elasticsearch.xpack.inference.services.sagemaker.model.SageMakerModel;
+
+import java.util.EnumSet;
+
+public interface SageMakerStreamSchemaPayload extends SageMakerSchemaPayload {
+    InferenceServiceResults.Result streamResponseBody(SageMakerModel model, SdkBytes response) throws Exception;
+
+    SdkBytes unifiedRequestBytes(SageMakerModel model, UnifiedCompletionRequest request) throws Exception;
```

Comment:
nit: We need to go through and change the naming of stuff to be a bit more consistent. I think we were leaning towards using chat completion instead of unified but I'm open to suggests.